### ISSUE TITLE
TagBamWithReadSequenceExtended with multiple cell barcodes

### ISSUE THREAD

**Issue by fbrundu** [broadinstitute/Drop-seq#41]
Hi,
I would like to use the tool __TagBamWithReadSequenceExtended__ for a technology with multiple cell barcodes. Is it possible to use it with an "append" behavior? For example, one run for the first round of barcodes, then subsequent runs on the other barcodes - appending the new barcodes to the tag XC?

Thanks for your work!
Francesco

**Comment by alecw** [comment]
Hi @fbrundu ,

Currently there is not an append option.  However, if the multiple cell barcodes are on the same end of the read pair, you can specify multiple ranges with the BASE_RANGE option.  Would that do the trick?

-Alec

**Comment by fbrundu** [comment]
Hi @alecw,
Thanks! Do you mean something like:

```shell
$dropseq_dir/TagBamWithReadSequenceExtended \
    I=$sampleName'.unaligned.sorted.bam' O=/dev/stdout \
    SUMMARY=$sampleName'.cell_tag_report.txt' \
    BASE_RANGE=11-19 BASE_RANGE=49-56 BASE_RANGE=87-94 \
    BASE_QUALITY=10 BARCODED_READ=1 DISCARD_READ=False \
    TAG_NAME=XC NUM_BASES_BELOW_QUALITY=1
```

Are the different ranges joined together in a single annotation?

Francesco

**Comment by jamesnemesh** [comment]
Yes. I believe someone in our lab explored split cell/molecular barcode chemistry at some point in the past.

-Jim
